### PR TITLE
fill port

### PR DESCRIPTION
**Pull request by hxrts**
I pulled out the facets fill work and touched a few things up for a clean merge. It doesn't touch any downstream files so we should be able to merge into master rather than RC without issue.

### CODE REVIEW

File: copy_number/facetsFillGeneCN.R [hunk 1]
```diff
@@ -154,8 +168,8 @@ OverCall <- function(column.name) {
             mutate(index=as.numeric(index)) %>%
             group_by(index) %>%
```
Comment (raylim) [inline copy_number/facetsFillGeneCN.R:171]: why do you `group_by` and then `ungroup` immediately after?

File: copy_number/facetsFillGeneCN.R [hunk 2]
```diff
@@ -186,16 +200,18 @@ cnv.matrix %<>%
 
 chrom <- cnv.matrix$chrom
 
-PlotCNHeatmap(cnv.matrix, file.name='facets/gene.cn.pdf', threshold=TRUE)
+PlotCNHeatmap(cnv.matrix, file.name='facets/geneCN.pdf', threshold=TRUE)
```
Comment (raylim) [inline copy_number/facetsFillGeneCN.R:210]: should output this to the specified output argument file (not hard-coded)

File: copy_number/facetsFillGeneCN.R [hunk 3]
```diff
@@ -265,24 +280,22 @@ for (sample.name in names(cnv.matrix) %>% list.filter(!. %in% c('chrom', 'start'
...
-cnv.matrix %<>% mutate(mid=(start+end)/2) %>% select(chrom, start, mid, end, everything()) %>% ungroup
-
 
-# plot final heatmap
-PlotCNHeatmap(cnv.matrix, file.name='facets/gene.cn.fill.pdf', threshold=TRUE)
+# add midpoints
+cnv.matrix %<>% mutate(mid=(start+end)/2) %>% select(chrom, start, mid, end, everything()) %>% ungroup
 
 # call overcall fix function
 names(cnv.matrix) %>%
```
Comment (raylim) [inline copy_number/facetsFillGeneCN.R:296]: weird code

File: copy_number/facetsFillGeneCN.R [hunk 4]
```diff
@@ -265,24 +280,22 @@ for (sample.name in names(cnv.matrix) %>% list.filter(!. %in% c('chrom', 'start'
...
 list.filter(!. %in% c('chrom', 'start', 'mid', 'end', 'hgnc', 'band')) %>%
 list.map(., .) %>%
 map(~ OverCall(.x))
 
+# plot final heatmap
+PlotCNHeatmap(cnv.matrix, file.name='facets/geneCN.fill.pdf', threshold=TRUE)
```
Comment (raylim) [inline copy_number/facetsFillGeneCN.R:302]: should output to specified file

File: copy_number/facetsFillGeneCN.R [hunk 5]
```diff
@@ -1,5 +1,8 @@
 #!/usr/bin/env Rscript
 # fill facets gene CN file
+
+suppressPackageStartupMessages(library("facets", lib.loc="/home/bermans/R-dev/"))
```
Comment (raylim) [inline copy_number/facetsFillGeneCN.R]: hard coded library location

Comment (hxrts) [inline copy_number/facetsFillGeneCN.R]: this script shouldn't need the facets library at all actually. I think that was just copied over earlier.

File: copy_number/facetsFillGeneCN.R (hunk 1 above)
Comment (hxrts) [inline copy_number/facetsFillGeneCN.R:171]: I'm grouping by the unique row index, left joining to each group, and then selecting the closest row from the second df with slice().

File: copy_number/facetsFillGeneCN.R (hunk 2 above)
Comment (hxrts) [inline copy_number/facetsFillGeneCN.R:210]: cool, will change.

File: copy_number/facetsFillGeneCN.R (hunk 3 above)
Comment (hxrts) [inline copy_number/facetsFillGeneCN.R:296]: yeah, not my favorite either.

**Pull request merged**